package/gluon-status-page-api/src/stations.c

```diff
@@ -0,0 +1,99 @@
+#include <stdio.h>
+#include <string.h>
+#include <unistd.h>
+#include <stdbool.h>
+#include <json.h>
+#include <iwinfo.h>
+#include <net/if.h>
+
+#define STR(x) #x
+#define XSTR(x) STR(x)
+
+static struct json_object *get_stations(const struct iwinfo_ops *iw, const char *ifname) {
+  int len;
+  char buf[IWINFO_BUFSIZE];
+  struct json_object *stations = json_object_new_object();
+
+  if (iw->assoclist(ifname, buf, &len) == -1)
+    return stations;
+
+  if (len <= 0)
+    return stations;
+
+  len /= sizeof(struct iwinfo_assoclist_entry);
+
+  struct iwinfo_assoclist_entry *entry = (struct iwinfo_assoclist_entry *)buf;
+
+  for (int i = 0; i < len; i++, entry++) {
```

Comment:
The variable `i` can be completely removed if the condition is changed to something like `(char*)(entry+1) <= buf+len`. The early return statement can be removed as the `for` condition will never be true if `len <= 0`.